CountItems/utils/utils.js

```diff
@@ -9,79 +9,111 @@ function consolidateDataMetrics(target, source) {
-                nonCurrent: 0,
-                _currentCold: 0,
-                _nonCurrentCold: 0,
-                _currentRestored: 0,
-                _currentRestoring: 0,
-                _nonCurrentRestored: 0,
-                _nonCurrentRestoring: 0,
-                _inflightsPreScan: 0,
-                _incompleteMPUParts: 0,
+                current: 0n,
+                nonCurrent: 0n,
+                _currentCold: 0n,
+                _nonCurrentCold: 0n,
+                _currentRestored: 0n,
+                _currentRestoring: 0n,
+                _nonCurrentRestored: 0n,
+                _nonCurrentRestoring: 0n,
+                _inflightsPreScan: 0n,
+                _incompleteMPUParts: 0n,
             },
         });
     }
     if (!resTarget.objectCount) {
         Object.assign(resTarget, {
             objectCount: {
-                current: 0,
-                nonCurrent: 0,
-                _currentCold: 0,
-                _nonCurrentCold: 0,
-                _currentRestored: 0,
-                _currentRestoring: 0,
-                _nonCurrentRestored: 0,
-                _nonCurrentRestoring: 0,
-                _incompleteMPUUploads: 0,
-                deleteMarker: 0,
+                current: 0n,
+                nonCurrent: 0n,
+                _currentCold: 0n,
+                _nonCurrentCold: 0n,
+                _currentRestored: 0n,
+                _currentRestoring: 0n,
+                _nonCurrentRestored: 0n,
+                _nonCurrentRestoring: 0n,
+                _incompleteMPUUploads: 0n,
+                deleteMarker: 0n,
             },
         });
     }
     if (!source) {
         return resTarget;
     }
     const { usedCapacity, objectCount, accountOwnerID } = source;
-    resTarget.usedCapacity.current += usedCapacity && usedCapacity.current ? usedCapacity.current : 0;
-    resTarget.usedCapacity.nonCurrent += usedCapacity && usedCapacity.nonCurrent ? usedCapacity.nonCurrent : 0;
-    resTarget.usedCapacity._currentCold += usedCapacity && usedCapacity._currentCold ? usedCapacity._currentCold : 0;
-    resTarget.usedCapacity._nonCurrentCold += usedCapacity && usedCapacity._nonCurrentCold ? usedCapacity._nonCurrentCold : 0;
-    resTarget.usedCapacity._currentRestoring += usedCapacity && usedCapacity._currentRestoring ? usedCapacity._currentRestoring : 0;
-    resTarget.usedCapacity._currentRestored += usedCapacity && usedCapacity._currentRestored ? usedCapacity._currentRestored : 0;
-    resTarget.usedCapacity._nonCurrentRestoring += usedCapacity && usedCapacity._nonCurrentRestoring ? usedCapacity._nonCurrentRestoring : 0;
-    resTarget.usedCapacity._nonCurrentRestored += usedCapacity && usedCapacity._nonCurrentRestored ? usedCapacity._nonCurrentRestored : 0;
-    resTarget.usedCapacity._incompleteMPUParts += usedCapacity && usedCapacity._incompleteMPUParts ? usedCapacity._incompleteMPUParts : 0;
+    resTarget.usedCapacity.current += BigInt(usedCapacity?.current || 0n);
```

Comment:
not sure we should take the chance and remove these, but it seems to me these `BigInt` conversions should not be useful : since these fields are only managed by `_processEntryData()` (which already handles bigint) and this function...

Reply:
I tried, but we end up with functional tests failures, maybe @benzekrimaha you will have a bit more info on this?

Reply:
I think it's safe and it's more like we don't pass the right values to this function in the test, so not in the actual program, having passing tests would be rethink them. Better be extra-safe here, as this is the most sensitive part of our changes.